File: 2024 January POTD/Smallest window containing 0, 1 and 2.java
class Solution {
    public int smallestSubstring(String S) {
        // Code here
        
        int char0=-1,char1=-1,char2 =-1;
        int ans=S.length();
        int flag = -1;
        for(int i=0;i<S.length();i++){
            if(S.charAt(i)=='0'){
                char0=i;
            }
            else if(S.charAt(i)=='1'){
                char1=i;
            }
            else if(S.charAt(i)=='2'){
                char2=i;
            }
            
            if(char0 !=-1 & char1!=-1 & char2!=-1){
                ans = Math.min(ans, i-Math.min(char0,Math.min(char1,char2))+1);
                flag = ans;
            }
        }
        return flag;
        
    }
}

File: 2026 April POTD/Smallest window containing 0, 1 and 2.cpp
class Solution {
public:
    int smallestSubstring(string s) {
        int count[10] = {0};
        int mini = INT_MAX;
        int j = 0;
        int cnt = 0;
        for(int i = 0;i<s.length();i++){
            count[s[i]-'0']++;
            if(count[s[i]-'0'] == 1){
                cnt++;
            }
            if(cnt == 3){
                mini = min(mini,i-j+1);
                while(1){
                   count[s[j]-'0']--;
                   if(count[s[j]-'0'] == 0){
                       mini = min(mini,i-j+1);
                       j++;
                       cnt--;
                       break;
                   }
                   j++;
                }
            }
        }
        if(mini == INT_MAX){
            return -1;
        }
        return mini;
    }
};
